operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/ResourceOwner.java

```diff
@@ -0,0 +1,26 @@
+package io.javaoperatorsdk.operator.processing;
+
+import java.util.Optional;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+public interface ResourceOwner<R, P extends HasMetadata> {
```

Comment:
Why was this extracted? I don't think it should be, what is an owner? 
basically this is reader, but that should always be, a DependentResource should always read the object or tell if the object does not exists

Reply:
Sorry, I misread it, for some reason I though its implemented by `AbstractDependentResource` not by dependent resource. ( I probably should close it for today :) )